src/channeldef.ts

```diff
@@ -198,6 +199,72 @@ export type TextDef<F extends Field> =
   | FieldOrDatumDefWithCondition<StringDatumDef<F>, Text>
   | ValueDefWithCondition<StringFieldDef<F>, Text>;
 
+/**
+ * Position of a label relative to its base mark's position, which includes anchor direction (anchor) and distance (offset).
+ */
+export type LabelPosition = {
+  /**
+   * distance of a label from its base mark.
+   * if negative, the label position is insize.
+   */
+  offset: number;
+
+  /**
+   * anchor direction of a label from its base mark.
+   * The value could be one of `'left '` , `'right '` , `'top '` , `'middle '` ,
+   * `'bottom '` , `'top-left '` , `'top-right '` , `'bottom-left '` , or  '`bottom-right '`
+   */
+  anchor: LabelAnchor;
+};
+
+export type LabelDefMixins = {
+  /**
+   * A list of possible positions for a label to be placed relative to its base mark.
+   */
+  position?: LabelPosition[];
+
+  /**
+   * The layers of marks that the labels will avoid.
+   * `'all'` means all the marks that are children of the root layer.
+   * number `n` means all the marks that are children of the nth ancestor layer.
+   *
+   * __Default value:__ `0` (only avoid the base mark)
+   */
+  avoidParentLayer?: 'all' | number;
+
+  /**
+   * Mark definition of the label text mark without the `'type'` property.
+   *
+   * __See also:__ [`text` mark](https://vega.github.io/vega-lite/docs/text.html#properties)
+   */
+  mark?: Omit<MarkDef<'text'>, 'type'>;
+
+  /**
+   * The padding in pixels by which a label may extend past the chart bounding box.
+   */
+  padding?: number;
+
+  /**
+   * The labeling method to use for area marks.
+   * One of `'floodfill'`, `'reduced-search'`, and `'naive'`.
```

Comment:
I think we need to explain more what these options means and when should they use them?